libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -284,6 +284,35 @@ bool GCS_MAVLINK::send_battery_status()
     return true;
 }
 
+void GCS_MAVLINK::send_battery_extended_status(const uint8_t instance) const
+{
+    const AP_BattMonitor &battery = AP::battery();
+    mavlink_msg_battery_extended_status_send(
+        chan,
+        instance, // id
+        battery.get_resistance(instance)
+    );
+}
+
+// returns true if all battery instances were reported
+bool GCS_MAVLINK::send_battery_extended_status()
+{
+    const AP_BattMonitor &battery = AP::battery();
+
+    for(uint8_t i = 0; i < AP_BATT_MONITOR_MAX_INSTANCES; i++) {
+        const uint8_t battery_id = (last_battery_extended_status_idx + 1) % AP_BATT_MONITOR_MAX_INSTANCES;
+        if (battery.get_type(battery_id) == AP_BattMonitor::Type::NONE) {
+            last_battery_extended_status_idx = battery_id;
+            continue;
+        }
+        CHECK_PAYLOAD_SIZE(BATTERY_EXTENDED_STATUS);
+        send_battery_extended_status(battery_id);
+        last_battery_extended_status_idx = battery_id;
+        return true;
```

Comment:
doesn't this only report the first battery? comment is "returns true if all battery instances were reported" but seems to return true after first message sent.